backend/baseclass.pm

```diff
@@ -806,6 +807,10 @@ sub check_asserted_screen {
             $self->freeze_vm();
             return {waiting_for_needle => 1, filename => $self->write_img($img, $img_filename), candidates => $failed_candidates};
         }
+        if ($self->stall_detected) {
+            backend::baseclass::write_crash_file();
+            bmwqemu::mydie "assert_screen fails, but we detected a timeout in the process, so we abort";
```

Comment:
This code is called by check_screen too. I imagine it can be misleading for some to report always as assert_screen.
How about using `check_asserted_screen fails and we ...` as in warning above;